core/src/client/mod.rs

```diff
@@ -675,4 +703,540 @@ impl StarknetClient for StarknetClientImpl {
+                ether_tx.public_key = None;
+                ether_tx.block_hash = block_hash;
+                ether_tx.block_number = block_number;
+            }
+            StarknetTransaction::Deploy(deploy_tx) => {
+                // Extract relevant fields from InvokeTransactionV0 and convert them to the corresponding fields in EtherTransaction
+                ether_tx.hash =
+                    PrimitiveH256::from_slice(&deploy_tx.transaction_hash.to_bytes_be());
+                class_hash = deploy_tx.class_hash;
+                // Define gas_price data
+                ether_tx.gas_price = None;
+
+                ether_tx.creates = None;
+                // Extracting the public_key
+                ether_tx.public_key = None;
+                ether_tx.to = None;
+                ether_tx.block_hash = block_hash;
+                ether_tx.block_number = block_number;
+            }
+            StarknetTransaction::DeployAccount(deploy_account_tx) => {
+                ether_tx.hash =
+                    PrimitiveH256::from_slice(&deploy_account_tx.transaction_hash.to_bytes_be());
+                class_hash = deploy_account_tx.class_hash;
+
+                ether_tx.nonce = felt_to_u256(deploy_account_tx.nonce);
+                // TODO: Get from estimate gas
+                ether_tx.gas_price = None;
+                // Extracting the signature
+                ether_tx.r = felt_option_to_u256(deploy_account_tx.signature.get(0))?;
+                ether_tx.s = felt_option_to_u256(deploy_account_tx.signature.get(1))?;
+                ether_tx.v = felt_option_to_u256(deploy_account_tx.signature.get(2))?;
+                // Extracting the to address
+                ether_tx.to = None;
+                // Extracting the gas
+                ether_tx.gas = U256::from(0);
+                // Extracting the chain_id
+                ether_tx.chain_id = Some(CHAIN_ID.into());
+                // Extracting the standard_v
+                ether_tx.standard_v = U256::from(0);
+                // Extracting the public_key
+                ether_tx.public_key = None;
+                ether_tx.block_hash = block_hash;
+                ether_tx.block_number = block_number;
+            }
+        }
+        let kakarot_class_hash = FieldElement::from_hex_be(KAKAROT_CONTRACT_ACCOUNT_CLASS_HASH)
+            .map_err(|e| {
+                KakarotClientError::OtherError(anyhow::anyhow!(
+                    "Failed to convert Starknet block hash to FieldElement: {}",
+                    e
+                ))
+            })?;
+        let kakarot_starknet_address =
+            FieldElement::from_hex_be(KAKAROT_CONTRACT_ACCOUNT_CLASS_HASH).map_err(|e| {
+                KakarotClientError::OtherError(anyhow::anyhow!(
+                    "Failed to convert Starknet block hash to FieldElement: {}",
+                    e
+                ))
+            })?;
+        if class_hash == kakarot_class_hash {
```

Comment:
something is not quite right, what did you want to do here?

Reply:
This is basically to filter comparing the class_hash and then storing the "kakarot contract" on "ethers_tx.to"

Reply:
but the ether_tx.to is never kakarot, rather, it is the kakarot-evm smart contract targeted by execute_at_address